### PR TITLE
Improvement of the snp_store

### PR DESCRIPTION
**Pull request by rrahn**
This is a mirror of #1824 which hat some unresolved merge conflicts.

### CODE REVIEW

File: apps/snp_store/snp_store.cpp [hunk 1]
```diff
@@ -703,82 +734,161 @@ int detectSNPs(SNPCallingOptions<TSpec> &options)
...
             }
         }
+    }
     // Allocate one BamAlignmentRecord for each input BAM file.
     std::vector<BamAlignmentRecord> records(length(options.readFNames));
 
     /////////////////////////////////////////////////////////////////////
+    std::time_t t = std::time(nullptr);                                         //Get time and date for vcf header
+    char now[24];
+    strftime(now, sizeof(now), "%Y%m%d", std::localtime(&t));
```
Comment (rrahn) [inline apps/snp_store/snp_store.cpp]: Do you really write the timestamp of that file into the vcf file?
Is that common praxis and really necessary?
We can make tests with timestamps. At least not with the current system.
So you need make the timestamp writing optional (using a macro: like ```SEQAN_SNP_STORE_DISABLE_TIME_STAMP```) and let your cmake script add this, when ```SEQAN_BUILD_SYSTEM``` equals ```DEVELOP```, like:
```shell
if ("${SEQAN_BUILD_SYSTEM}" STREQUAL "DEVELOP")
   message (STATUS "Disabling time stamp feature for writing vcf files!")
   set (SEQAN_DEFINITIONS ${SEQAN_DEFINITIONS} -DSEQAN_SNP_STORE_DISABLE_TIME_STAMP)
endif ()
```
Thus you are able to fix the app tests.

File: apps/snp_store/snp_store.cpp [hunk 2]
```diff
@@ -798,94 +908,160 @@ int detectSNPs(SNPCallingOptions<TSpec> &options)
...
+//                                     tmpReads,
+//                                     tmpReadCounts,
+//                                     tmpReadCounts,
+//                                     tmpReadClips,
+//                                     tmpReadCigars)
+//                             firstprivate()
+//                             shared(options, sumWindows, i)
+//                             reduction(+:sumwindows)
+//                             schedule(dynamic, chunk)
+//                             ordered)
```
Comment (rrahn) [inline apps/snp_store/snp_store.cpp]: can this be removed

File: apps/snp_store/snp_store.cpp [hunk 3]
```diff
@@ -798,94 +908,160 @@ int detectSNPs(SNPCallingOptions<TSpec> &options)
...
+        for (currentWindowBegin = 0;
+             currentWindowBegin < (TContigPos)length(genomes[i]);
+             currentWindowBegin += options.windowSize)
         {
             TContigPos currentWindowEnd = currentWindowBegin + options.windowSize;
-            if(currentWindowEnd > (TContigPos)length(genomes[i])) currentWindowEnd = (TContigPos)length(genomes[i]);
-
-            if(positionStatsOnly && !(currentWindowBegin <= *inspectPosIt && *inspectPosIt< currentWindowEnd))
+            if (currentWindowEnd > (TContigPos)length(genomes[i]))
+                currentWindowEnd = (TContigPos)length(genomes[i]);
```
Comment (rrahn) [inline apps/snp_store/snp_store.cpp]: add surrounding ```{}```

File: apps/snp_store/snp_store.cpp [hunk 4]
```diff
@@ -938,103 +1128,177 @@ int detectSNPs(SNPCallingOptions<TSpec> &options)
...
+                //                if (*options.outputCNV != 0)
+                //                {
+                //                    dumpCopyNumberPolymorphismsBatch(fragmentStore,
+                //                                                     genomeNames[i],
+                //                                                     startCoord,
+                //                                                     currentWindowBegin,
+                //                                                     currentWindowEnd,
+                //                                                     cnvFileStream,
+                //                                                     options);
+                //                }
```
Comment (rrahn) [inline apps/snp_store/snp_store.cpp]: can be removed? Otherwise document what it is for and when to add it.

**Comment by serosko** [comment]
I should manage to correct the issues today.

File: apps/snp_store/snp_store.cpp [hunk 5]
```diff
@@ -81,39 +96,42 @@ bool loadGenomes(TGenomeSet &genomes, StringSet<CharString> &fileNameList, ::std
...
         for (; !atEnd(seqFileIn); ++seqCount)
         {
-            readRecord(temp, seq, seqFileIn);
+            try
+            {
+                readRecord(temp, seq, seqFileIn);
+            }
+            catch (ParseError const & e)
+            {
+                //std::cout << e.what() << "Base will be considered as 'N'." << std::endl;
```
Comment (serosko) [inline apps/snp_store/snp_store.cpp]: If I remember correctly, the standard output was flooded with error messages I used SNPStore on a FASTA file containing not only A,C,G,T,N.  I fixed it now, s.t. the warning will only be written to ouput when it occurs for the first time.

File: apps/snp_store/snp_store.cpp (hunk 1 above)
Comment (serosko) [inline apps/snp_store/snp_store.cpp]: I cannot compile or run the code where I am working from at the moment, hence I would almost certainly introduce some errors if i tried to switch to chrono library now.
As for the time stamp: Yes it is common to write it to the header of the vcf file (see https://samtools.github.io/hts-specs/VCFv4.2.pdf). I cannot change this now for the same reason as mentioned above.

File: apps/snp_store/snp_store.cpp (hunk 4 above)
Comment (serosko) [inline apps/snp_store/snp_store.cpp]: As it seems there endeavours to add copy-number-variation detection or something similar to SNP-Store. I never touched these parts of the code myself, but all of them seem to be commented out. My guess woul be that the might come in handy, if someone wants to add this functionality in the future.

**Comment by serosko** [comment]
I applied the suggested changes, and made a new pull request after merging everything with upstream/develop. Because I cannot compile or run the code where I am working from at the moment, I did not switch to chrono library and also did not change the behaviour of the time stamp. Consequently no newly generated test files were added.

File: apps/snp_store/snp_store.cpp (hunk 1 above)
Comment (rrahn) [inline apps/snp_store/snp_store.cpp]: @serosko any chance you can work on that before Monday?

**Comment by rrahn** [comment]
@serosko do you think you can do the last changes until Monday?

**Comment by serosko** [comment]
Sorry, I didn't check my emails during the weekend. Also I won't be able to make any changes until at least thursday.
________________________________
Von: rrahn [notifications@github.com]
Gesendet: Freitag, 9. Dezember 2016 20:00
An: seqan/seqan
Cc: Roskosch, Sebastian; Mention
Betreff: Re: [seqan/seqan] Improvement of the snp_store (#1961)


@serosko<https://github.com/serosko> do you think you can do the last changes until Monday?

**Comment by rrahn** [comment]
closed in favor of #1962

**Pull request closed without merging**